src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -929,11 +920,53 @@ export class SearchSource {
       delete searchRequest.highlightAll;
     }
 
-    if (pit) {
-      body.pit = pit;
-    }
+    const omitByIsNil = (object: Record<string, any>) => omitBy(object, isNil);
 
-    return searchRequest;
+    const bodyToReturn = {
+      ...searchRequest.body,
+      pit: searchRequest.pit,
+    };
+
+    return omitByIsNil({ ...searchRequest, body: omitByIsNil(bodyToReturn) }) as SearchRequest;
+  }
+
+  private getUniqueFields({
+    index,
+    fields,
+    metaFields,
+    filteredDocvalueFields,
+  }: {
+    index?: DataView;
+    fields: any;
+    metaFields: string;
+    filteredDocvalueFields: any;
+  }) {
+    const bodyFields = this.getFieldsWithoutSourceFilters(index, fields);
+    // if items that are in the docvalueFields are provided, we should
+    // inject the format from the computed fields if one isn't given
+    const docvaluesIndex = keyBy(filteredDocvalueFields, 'field');
+
+    const uniqueFieldNames = new Set();
+    const uniqueFields = [];
+    for (const field of bodyFields.concat(filteredDocvalueFields)) {
+      const fieldName = this.getFieldName(field);
+      if (metaFields.includes(fieldName) || uniqueFieldNames.has(fieldName)) {
+        continue;
+      }
+      uniqueFieldNames.add(fieldName);
+      if (Object.keys(docvaluesIndex).includes(fieldName)) {
```

Comment:
Can `Object.keys(docvaluesIndex)` be evaluated once? Or maybe build a `Set` out of its keys

Reply:
I'll address this once the PR goes green